src/atoms/atomgraph.jl

```diff
@@ -48,38 +48,33 @@ end
 
 
 # initialize directly from adjacency matrix
-AtomGraph(adj::Array{R}, elements::Vector{String}, id = "") where {R<:Real} =
-    AtomGraph(SimpleWeightedGraph{R}(adj), elements, id)
-
-
 AtomGraph(adj::Array{R}, elements::Vector{String}, id = "") where {R<:Real} =
     AtomGraph(SimpleWeightedGraph(adj), elements, id)
 
 """
-    AtomGraph(input_file_path; id="", output_file_path=nothing, featurization=nothing, overwrite_file=false, use_voronoi=false, cutoff_radius=8.0, max_num_nbr=12, dist_decay_func=inverse_square, normalize_weights=true)
+    AtomGraph(input_file_path, id = splitext(input_file_path)[begin]; output_file_path = nothing, featurization = nothing, overwrite_file = false, use_voronoi = false, cutoff_radius = 8.0, max_num_nbr = 12, dist_decay_func = inverse_square, normalize_weights = true)
 
 Construct an AtomGraph object from a structure file.
 
 # Required Arguments
 - `input_file_path::String`: path to file containing structure (must be readable by ASE.io.read)
+- `id::String`: ID associated with structure (e.g. identifier from online database). Defaults to name of input file if undefined.
```

Comment:
Maybe, but then I'm not sure how you'd be able to do something like this -
https://github.com/Chemellia/ChemistryFeaturization.jl/blob/99618bcb3e3eb5828ea4efd84ede5ea627dc4e3d/test/atoms/AtomGraph_tests.jl#L64

Reply:
I just meant in the docstring – this might just be a vocabulary thing. I think of the definitions as follows:
* required: user must provide (opposite of optional)
* optional: either not necessary or comes with a default value so user doesn't need to provide a value
* positional: defined only by its position in argument list (opposite of keyword arg)
* keyword: defined by the syntax `key=val`, doesn't need to appear in a particular order
But maybe you're thinking of required == optional?

Reply:
Oh right, then it makes sense.

> But maybe you're thinking of required == optional?
No, I was thinking `optional == keyword`, now I realize that need not always be the case.